x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/rules_table_filters/rule_execution_status_selector.tsx

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState } from 'react';
+import type { EuiSelectableOption } from '@elastic/eui';
+import { EuiFilterButton, EuiPopover, EuiSelectable } from '@elastic/eui';
+import * as i18n from '../../../../../detections/pages/detection_engine/rules/translations';
+import { RuleExecutionStatus } from '../../../../../../common/detection_engine/rule_monitoring/model/execution_status';
+import { getCapitalizedStatusText } from '../../../../../detections/components/rules/rule_execution_status/utils';
+import { RuleStatusBadge } from '../../../../../detections/components/rules/rule_execution_status/rule_status_badge';
+
+interface OptionData {
+  status: RuleExecutionStatus;
+}
+
+interface RuleExecutionStatusSelectorProps {
+  selectedStatus?: RuleExecutionStatus;
+  onSelectedStatusChanged: (newStatus?: RuleExecutionStatus) => void;
+}
+
+/**
+ * Selector for selecting last rule execution status to filter on
+ *
+ * @param selectedStatus Selected rule execution status
+ * @param onSelectedStatusChanged change listener to be notified when rule execution status selection changes
+ */
+const RuleExecutionStatusSelectorComponent = ({
+  selectedStatus,
+  onSelectedStatusChanged,
+}: RuleExecutionStatusSelectorProps) => {
+  const [isExecutionStatusPopoverOpen, setIsExecutionStatusPopoverOpen] = useState(false);
+
+  const selectableOptions = [
+    {
+      label: getCapitalizedStatusText(RuleExecutionStatus.succeeded),
+      data: { status: RuleExecutionStatus.succeeded },
+      checked: selectedStatus === RuleExecutionStatus.succeeded ? 'on' : undefined,
+    },
+    {
+      label: getCapitalizedStatusText(RuleExecutionStatus['partial failure']),
+      data: { status: RuleExecutionStatus['partial failure'] },
+      checked: selectedStatus === RuleExecutionStatus['partial failure'] ? 'on' : undefined,
+    },
+    {
+      label: getCapitalizedStatusText(RuleExecutionStatus.failed),
+      data: { status: RuleExecutionStatus.failed },
+      checked: selectedStatus === RuleExecutionStatus.failed ? 'on' : undefined,
+    },
+  ] as EuiSelectableOption[];
```

Comment:
Let's avoid casting via `as` whenever we can. Here's a way to make it type-safe:

```ts
  const selectableOptions: EuiSelectableOption[] = [
    {
      label: getCapitalizedStatusText(RuleExecutionStatus.succeeded) ?? '',
      data: { status: RuleExecutionStatus.succeeded },
      checked: selectedStatus === RuleExecutionStatus.succeeded ? 'on' : undefined,
    },
    {
      label: getCapitalizedStatusText(RuleExecutionStatus['partial failure']) ?? '',
      data: { status: RuleExecutionStatus['partial failure'] },
      checked: selectedStatus === RuleExecutionStatus['partial failure'] ? 'on' : undefined,
    },
    {
      label: getCapitalizedStatusText(RuleExecutionStatus.failed) ?? '',
      data: { status: RuleExecutionStatus.failed },
      checked: selectedStatus === RuleExecutionStatus.failed ? 'on' : undefined,
    },
  ];
```